### PR TITLE
Move URL Checker to Weekly Job

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
We want URL checking, but we don't want it to cripple normal jobs. This moves the URL checker to happen weekly on Sunday.

## Changes proposed in this PR:
- Remove URL checker from regular test jobs
- Create new weekly / Sunday night URL checker

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: .github/workflows/url_check.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        file_types: .md,.rst,.py
+        # Choose whether to include file with no URLs in the prints.
+        print_all: false
+        # More verbose summary at the end of a run
+        verbose: true
+        # How many times to retry a failed request (defaults to 1)
+        retry_count: 3
+        # Exclude:
+        #  - Jenkins because it's behind a firewall
+        #  - RTD because a magically-generated string triggers failures
```
Comment (blnicho) [inline .github/workflows/url_check.yml:31]: Is this comment out of date?

Comment (mrmundt) [reply .github/workflows/url_check.yml:31]: Nope, still correct.

**Review by jsiirola — APPROVED** [review]

File: .github/workflows/test_branches.yml [hunk 2]
```diff
@@ -47,20 +47,6 @@ jobs:
       uses: crate-ci/typos@master
       with: 
         config: ./.github/workflows/typos.toml
-    - name: URL Checker
```
Comment (jsiirola) [inline .github/workflows/test_branches.yml:50]: Should we leave this here - either as is, or as a separate job?  That way Devs would be informed about failing URLs as part of their dev process, but it wouldn't hold up the PR builds / reviews?

Comment (blnicho) [reply .github/workflows/test_branches.yml:50]: I like this idea for catching issues with new URLs being added

**Comment by mrmundt** [comment]
@jsiirola , @blnicho - added URL checking back to the branch job

**Review by jsiirola — APPROVED** [review]

**Pull request merged**